test/jdk/javax/swing/JButton/HtmlButtonImageTest/HtmlButtonImageTest.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright 2022 JetBrains s.r.o.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @key headful
+ * @bug 8015854
+ * @summary JButton text set as html image had additional unwanted padding
+ * @run main HtmlButtonImageTest
+ */
+
+import java.awt.Dimension;
+import java.awt.FlowLayout;
+import java.awt.Point;
+import java.awt.Insets;
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.image.BufferedImage;
+import java.awt.Robot;
+import java.net.URL;
+import java.nio.file.Path;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.SwingConstants;
+import javax.swing.UnsupportedLookAndFeelException;
+import javax.imageio.ImageIO;
+import java.io.IOException;
+import java.io.File;
+
+
+public final class HtmlButtonImageTest {
+    private static JFrame frame;
+    private static Point point;
+    private static URL urlImage;
+    private static JButton button;
+
+    private static Path srcDir;
+
+    public static final int BUTTON_HEIGHT = 37;
+    public static final int BUTTON_WIDTH = 37;
+    public static final int SQUARE_HEIGHT = 19;
+    public static final int SQUARE_WIDTH = 19;
+    public static final int PIXEL_BUFFER = 3;
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        robot.setAutoDelay(2000);
+        robot.setAutoWaitForIdle(true);
+
+        // store path to source directory to locate image
+        srcDir = Path.of(System.getProperty("test.src", "."));
+
+        // generate red_square.png image to use as JButton text
+        SwingUtilities.invokeAndWait(() -> {
```

Comment:
generateImage isn't doing anything that needs to be run on the EDT. Get rid of invokeAndWait here.